DOCSP-40417: Add X509 troubleshooting steps (#213)
mongodb/docs-csharp: source/connection-troubleshooting.txt

instance hosted on the local machine with the same code. A deployment on
the same machine doesn't require any authorization to connect.

.. _csharp-troubleshooting-connection-X509:

Check Your Ephemeral Key Usage

> ------------------------------

Reviewer: The header for this section only discusses `SCRAM` authentication, so I think you can either add a small intro for x.509 errors to the intro or you can bump this section up a header and add some introductory information.

> There is a {+language+} bug on Windows where the {+driver-short+} may be unable to locate an
> ``X.509`` authentication certificate created in memory. This error is accompanied by the following
> error message:

Reviewer: ```suggestion
If you are using Windows as your operating system, you might encounter an issue 
where the {+driver-short+} is unable to locate an ``X.509`` authentication certificate created
in memory. This error is raised with the following error message:
```

.. code-block:: none

   No credentials are available in the security package

> This is a known Windows OS issue where authentication using ephemeral keys may fail. As a
> workaround, use the following code to generate any ``X.509`` certificates required by your
> application:

Reviewer: ```suggestion
To resolve this error, use the following code to generate an ``X.509`` certificates required by your
application:
```

instance hosted on the local machine with the same code. A deployment on
the same machine doesn't require any authorization to connect.

.. _csharp-troubleshooting-connection-X509:

X.509 Credential Error
~~~~~~~~~~~~~~~~~~~~~~

> If you are using Windows as your operating system, you might encounter an issue where the

Reviewer: ```suggestion
If you are using Windows as your operating system, you might encounter an issue in which the
```

{+driver-short+} is unable to locate an ``X.509`` authentication certificate in memory.
This error is raised with the following error message:

> .. code-block:: none

Reviewer: ```suggestion
.. code-block:: none
   :copyable: false
```
